satpy/demo/__init__.py

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2019 SatPy developers
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""Demo data download helper functions.
+
+Each ``get_*`` function below downloads files to a local directory and returns
+a list of paths to those files. Some (not all) functions have multiple options
+for how the data is downloaded (via the ``method`` keyword argument)
+including:
+
+- gcsfs: Download data from a public google cloud storage bucket using the
+    ``gcsfs`` package.
+- unidata_thredds: Access data using OpenDAP or similar method from Unidata's
+    public THREDDS server
+    (https://thredds.unidata.ucar.edu/thredds/catalog.html).
+- uwaos_thredds: Access data using OpenDAP or similar method from the
+    University of Wisconsin - Madison's AOS department's THREDDS server.
+- http: A last resort download method when nothing else is available of a
+    tarball or zip file from one or more servers available to the SatPy
+    project.
+- uw_arcdata: A network mount available on many servers at the Space Science
+    and Engineering Center (SSEC) at the University of Wisconsin - Madison.
+    This is method is mainly meant when tutorials are taught at the SSEC
+    using a Jupyter Hub server.
+
+To use these functions, do:
+
+    >>> from satpy import Scene, demo
+    >>> filenames = demo.get_us_midlatitude_cyclone_abi()
+    >>> scn = Scene(reader='abi_l1b', filenames=filenames)
```

Comment:
I think it would be more intuitive for the user if `demo.get...` would return the same thing as `find_files_and_readers`, ie that the user won't have to specify the reader.

Reply:
I thought about that. My arguments to myself when thinking about this are:

1. The same files could theoretically be useful to multiple readers. Best example of this I can think of is a VIIRS EDR reader that needs VIIRS SDR geolocation and/or some of the SDR bands.
2. Doing it the way I have it resembles the usual use case IMO. It requires the user to know or at least see what reader they are using. Hiding that in the `filenames` by using it as a dictionary may confuse new users.

Willing to be convinced otherwise.